cparser/Elab.ml

```diff
@@ -1769,16 +1823,22 @@ let elab_expr vararg loc env a =
       begin match elab env (BINARY(sop, a1, a2)) with
       | ({ edesc = EBinop(_, b1, b2, _); etyp = ty } as b),env  ->
           if List.mem AConst (attributes_of_type env b1.etyp) then
-            err "left-hand side of assignment has 'const' type";
+            err "read-only variable is not assignable";
           if not (is_modifiable_lvalue env b1) then
-            err ("left-hand side of assignment is not a modifiable l-value");
+            err "expression is not assignable";
           if not (wrap2 valid_assignment loc env b b1.etyp) then begin
             if wrap2 valid_cast loc env ty b1.etyp then
-              warning "assigning a value of type@ %a@ to a lvalue of type@ %a"
-                      Cprint.typ ty Cprint.typ b1.etyp
+              if wrap2 int_pointer_conversion loc env ty b1.etyp then
+                warning Int_conversion
+                  "incompatible integer to pointer conversion assigning to %a from %a"
+                   (print_typ env) b1.etyp  (print_typ env) ty
+              else
+                warning Int_conversion
+                  "incompatible pointer to integer conversion assgining to %a from %a"
+                  (print_typ env) b1.etyp (print_typ env) ty
             else
```

Comment:
See previous comment: these are not the only two cases.  Plus, that's a lot of duplicated code!  Please factor this error reporting in a separate function.  Or give up on reporting specially int->pointer and pointer->int conversions.  (Why single them out?)